src/endpoint/s3/ops/s3_put_bucket_lifecycle.js

```diff
@@ -108,11 +109,16 @@ async function put_bucket_lifecycle(req) {
         }
         id_set.add(current_rule.id);
 
-        if (rule.Status?.length !== 1) {
+        if (!rule.Status || rule.Status.length !== 1) {
             dbg.error('Rule should have status', rule);
             throw new S3Error(S3Error.InvalidArgument);
```

Comment:
Since you touch it, I would check that it is `InvalidArgument` (or open a gap to validate it).

Reply:
The first error is from the AWS client as I understand it (not from the server).
I agree with your thought.
Anyway, in the current code we can see it is `InvalidArgument` and in the screenshot it is `MalformedXML`.

Reply:
I think aws cli itself preventing the api call if the policy doesn't contains the required params here.